tests/contrib/kafka/test_kafka.py

```diff
@@ -618,3 +618,44 @@ def _read_single_message(consumer):
     consumer.commit(asynchronous=False)
     assert consumer.committed([TopicPartition(kafka_topic, 0)])[0].offset == 2
     assert list(buckets.values())[0].latest_commit_offsets[ConsumerPartitionKey("test_group", kafka_topic, 0)] == 1
+
+
+def test_data_streams_kafka_default_context_propagation(dummy_tracer, consumer, producer, kafka_topic):
```

Comment:
Split this test into two: one for DSM (pathway is propagated) and one for APM (traces are not equal).

The APM test should be named something like "test_context_is_not_propagated" with a comment explaining why (basically it isn't the current behavior)

They are separate products so we should split out the tests.